apps/passport-server/src/services/poapService.ts

```diff
@@ -205,13 +287,54 @@ export class PoapService {
     }
   }
 
+  /**
+   * Given a ZKEdDSAEventTicketPCD sent to the server for claiming a Zuzalu 2023 POAP,
+   * returns the valid redirect URL to the response handler.
+   *  1. If this ticket is already associated with a POAP mint link, return that link.
+   *  2. If this ticket is not associated with a POAP mint link and more unclaimed POAP
+   *     links exist, then associate that unclaimed link with this ticket and return it.
+   *  3. If this ticket is not associated with a POAP mint link and no more unclaimed
+   *     POAP links exist, return a custom server error URL.
+   */
+  public async getZuzalu23PoapRedirectUrl(
+    serializedPCD: string
+  ): Promise<string> {
+    try {
+      const ticketId = await this.validateZuzalu23Ticket(serializedPCD);
+      const poapLink = await this.getPoapClaimUrlByTicketId(
+        ticketId,
+        "zuzalu23"
+      );
+      if (poapLink == null) {
+        throw new Error("Not enough Zuzalu 2023 POAP links");
+      }
+      return poapLink;
+    } catch (e) {
+      logger("[POAP] getZuzalu23PoapRedirectUrl error", e);
+      this.rollbarService?.reportError(e);
+      // Return the generic /server-error page instead for the route to redirect to,
+      // with a title and description informing the user to contact support.
+      return getServerErrorUrl(
+        "Contact support",
+        "An error occurred while fetching your POAP mint link for Zuzalu 2023."
+      );
+    }
+  }
+
   /**
    * Helper function to handle the logic of retrieving the correct POAP mint link
-   * given the ticket ID. Returns NULL if the ticket is not associate with a link
-   * and no more unclaimed links exist.
+   * given the ticket ID.
+   *  1. If this ticket ID is already associated with a POAP mint link,
+   *     return that link.
+   *  2. If this ticket ID is not yet associated with a POAP mint link,
+   *     a new POAP mint link with the given poapEvent is associated with
+   *     the ticket ID, and that link is returned.
+   *  3. If this ticket ID is not associated with a POAP mint link and
+   *     no more unclaimed POAP mint links exist, return NULL.
    */
-  public async getDevconnectPoapClaimUrlByTicketId(
-    ticketId: string
+  public async getPoapClaimUrlByTicketId(
+    ticketId: string,
+    poapEvent: PoapEvent
   ): Promise<string | null> {
     return traced(
       "poap",
```

Comment:
you probably want to rename the `traced` span function name to match the new name of this function

Reply:
adding a `poapEvent` attribute inside the `traced` seems worth it imo too